test/e2e/autotls/config/dnssetup/main.go

```diff
@@ -77,14 +79,47 @@ func createDNSRecord(dnsRecord *config.DNSRecord) error {
 	return config.ChangeDNSRecord(addition, svc, env.CloudDNSProject, env.DNSZone)
 }
 
-func waitForDNSRecordVisibleLocally(record *config.DNSRecord) error {
+func waitForDNSRecordVisible(record *config.DNSRecord) error {
+	nameservers, err := net.LookupNS(env.DomainName)
+	if err != nil {
+		return fmt.Errorf("failed to lookup NS records for domain: %v", err)
+	}
+
 	return wait.PollImmediate(10*time.Second, 300*time.Second, func() (bool, error) {
-		ips, _ := net.LookupHost(record.Domain)
-		for _, ip := range ips {
-			if ip == record.IP {
-				return true, nil
+		for _, ns := range nameservers {
+			nsIP, err := net.LookupHost(ns.Host)
+			if err != nil {
+				return true, err
```

Comment:
If one of the host server does not have the DNS record yet, will this function return error? If it returns error, should we just log the error and return "false, nil" to trigger retry?

Reply:
This is looking up the IP for the host in the NS record. I'll relax this check and have it just retry.